### PR TITLE
Handle large octet-stream (master branch)

### PR DESCRIPTION
**Pull request by nathannau**
### Issue

In scenarios where the request body size exceeds the available RAM, the ESP fails to allocate sufficient memory to process the request in one go.

### Change

Now, when the "Content-Type" of the request is "application/octet-stream", it will attempt to call the "ufn" function of the handler, similar to how it handles multipart requests.

**Code Example:**

```cpp
server.on(uri, HTTP_PUT, finalCallback, [](){
  HTTPRaw raw = server.raw();
  if (raw.status == RAW_START) {
    ...
  } else if (raw.status == RAW_WRITE) {
    ...
  } else if (raw.status == RAW_END) {
    ...
  }
});
```

### CODE REVIEW

**Comment by lucasssvaz** [comment]
@nathannau Please pull the commits and don't force push over them. I had already fixed the CI errors for the example.

**Comment by lucasssvaz** [comment]
Following the instruction on the readme causes the error `curl: (55) Send failure: Broken pipe`.

**Review by lucasssvaz — CHANGES_REQUESTED** [review]
Example doesn't work as expected: `curl: (55) Send failure: Broken pipe`

**Comment by me-no-dev** [comment]
in the example folder you need to add an empty file with name `.skip.esp32h2`

**Review by lucasssvaz — CHANGES_REQUESTED** [review]

File: libraries/WebServer/src/detail/RequestHandlersImpl.h [hunk 1]
```diff
@@ -38,6 +38,12 @@ class FunctionRequestHandler : public RequestHandler {
 
         return true;
     }
+    bool canRaw(String requestUri) override {
+        if (!_ufn || _method == HTTP_GET)
+            return false;
```
Comment (lucasssvaz) [inline libraries/WebServer/src/detail/RequestHandlersImpl.h:43]:
```suggestion
        if (!_ufn || _method == HTTP_GET) {
            return false;
        }
```

File: libraries/WebServer/src/detail/RequestHandlersImpl.h [hunk 2]
```diff
@@ -55,6 +61,13 @@ class FunctionRequestHandler : public RequestHandler {
             _ufn();
     }
 
+    void raw(WebServer& server, String requestUri, HTTPRaw& raw) override {
+        (void)server;
+        (void)raw;
+        if (canRaw(requestUri))
+            _ufn();
```
Comment (lucasssvaz) [inline libraries/WebServer/src/detail/RequestHandlersImpl.h:68]:
```suggestion
        if (canRaw(requestUri)) {
            _ufn();
        }
```

File: libraries/WebServer/src/detail/RequestHandlersImpl.h (hunk 1 above)
Comment (me-no-dev) [reply libraries/WebServer/src/detail/RequestHandlersImpl.h:43]: What's the problem with that request @nathannau ?

**Review by lucasssvaz — CHANGES_REQUESTED** [review]
Please add an empty `.skip.esp32h2` file alongside your example

**Review by P-R-O-C-H-Y — APPROVED** [review]
Working as expected, tested also SDWebServer example with no issues.
@nathannau  Please add the skip file for H2. Thanks

**Pull request merged**